lib/Security/Encoder/Factory/UserAwareEncoderFactory.php

```diff
@@ -54,8 +54,17 @@ public function getEncoder($user)
             ));
         }
 
-        if (isset($this->encoders[$user->getUsername()])) {
-            return $this->encoders[$user->getUsername()];
+        $username = null;
```

Comment:
Heck, I would even rename the variable to `$userIdentifier` or `$identifier`, since `$username` is not applicable anymore. 😅